### PR TITLE
feat: add order to coder_parameter

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/8193

This PR adds `order` property to `coder_parameter` to determine the position of parameters presented in the UI/CLI form. More details can be found in the related issue.

### CODE REVIEW

File: provider/parameter.go [hunk 1]
```diff
@@ -331,6 +335,11 @@ func parameterDataSource() *schema.Resource {
 				Computed:    true,
 				Description: "Whether this value is optional.",
 			},
+			"priority": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The priority of template parameters determines their position ahead of parameters with lower priorities in the UI/CLI presentation.",
```
Comment (mafredri) [inline provider/parameter.go:341]: Maybe we can define the edge cases here a bit more?

```suggestion
				Description: "The priority of a template parameter determines its position in the UI/CLI presentation. A higher priority parameter is positioned ahead of a lower priority one and parameters with the same priority are ordered by name in ascending order.",
```

**Review by mafredri — APPROVED** [review]
Looks good 👍🏻. Do we need a test to check it as well?

File: provider/parameter.go (hunk 1 above)
Comment (mtojek) [reply provider/parameter.go]: Done!

**Comment by mtojek** [comment]
> Do we need a test to check it as well?

Done!

**Review by spikecurtis — APPROVED** [review]

**Comment by mtojek** [comment]
I adjusted the PR to [product expectations](https://github.com/coder/coder/issues/8193#issuecomment-1611149679).

**Review by mafredri — APPROVED** [review]

File: provider/parameter.go [hunk 2]
```diff
@@ -335,10 +335,10 @@ func parameterDataSource() *schema.Resource {
 				Computed:    true,
 				Description: "Whether this value is optional.",
 			},
-			"priority": {
+			"order": {
 				Type:        schema.TypeInt,
 				Optional:    true,
-				Description: "The priority of a template parameter determines its position in the UI/CLI presentation. A higher priority parameter is positioned ahead of a lower priority one and parameters with the same priority are ordered by name in ascending order.",
+				Description: "The order determines the position of a template parameter in the UI/CLI presentation. A parameter with lower order is positioned ahead of a high order one and parameters with the same order are sorted by name in ascending order.",
```
[the remark this answers is not in the document]
Comment (mtojek) [reply provider/parameter.go]: It sounds more natural, thanks!

**Pull request merged**